spark/common/src/main/scala/org/apache/spark/sql/sedona_sql/expressions/InferredExpression.scala

```diff
@@ -100,6 +100,10 @@ object InferrableType {
     new InferrableType[Array[Byte]] {}
   implicit val longArrayInstance: InferrableType[Array[java.lang.Long]] =
     new InferrableType[Array[java.lang.Long]] {}
+  implicit val intArrayInstance: InferrableType[Array[Int]] =
+    new InferrableType[Array[Int]] {}
+  implicit val javaIntArrayInstance: InferrableType[Array[java.lang.Integer]] =
```

Comment:
Why do you need both `Int` and `java.lang.Integer`?

Reply:
The existing approach is using Array[java.lang.Long]. Why not follow the same style?

Reply:
My bad, will remove the `Int` and `java.lang.Integer` and replace it with `java.lang.Long`

Reply:
Dude, what I mean is that why not use Array[java.lang.Integer] just like Array[java.lang.Long]

Reply:
No particular reason, scala by default takes `Int`. I guess the same reason we support `Double` and `java.lang.Double`

Reply:
OK got it @Kontinuation Can you confirm this?

Reply:
Scala `Int` is a primitive type, which is different from `java.lang.Integer`. It is necessary to have InferrableType instances for both `Array[Int]` and `Array[java.lang.Integer]` to support passing both `int` and `int[]` parameters.